atr/get/distribution.py

```diff
@@ -66,55 +68,116 @@ async def list_get(session: web.Committer, project: str, version: str) -> str:
+            shared.html_tr("Staging", "Yes" if dist.staging else "No"),
+            shared.html_tr("Upload date", str(dist.upload_date)),
+            shared.html_tr_a("API URL", dist.api_url),
+            shared.html_tr_a("Web URL", dist.web_url),
         ]
         block.table(".table.table-striped.table-bordered")[tbody]
-        form_action = util.as_url(post.distribution.delete, project=project, version=version)
-        delete_form_element = forms.render_simple(
-            delete_form,
-            action=form_action,
-            submit_classes="btn-danger",
-        )
-        block.append(htm.div(".mb-3")[delete_form_element])
+
+        # Create inline delete form with confirmation dialog (following projects.py pattern)
+        delete_form = htm.form(
+            ".d-inline-block.m-0",
+            method="post",
+            action=util.as_url(post.distribution.delete, project=project, version=version),
+            onsubmit=(
+                f"return confirm('Are you sure you want to delete the distribution "
+                f"{dist.platform.name} {dist.package} {dist.version}? This cannot be undone.');"
+            ),
+        )[
+            form.csrf_input(),
+            htpy.input(type="hidden", name="release_name", value=dist.release_name),
+            htpy.input(type="hidden", name="platform", value=dist.platform.name),
+            htpy.input(type="hidden", name="owner_namespace", value=dist.owner_namespace or ""),
+            htpy.input(type="hidden", name="package", value=dist.package),
+            htpy.input(type="hidden", name="version", value=dist.version),
+            htpy.button(".btn.btn-danger.btn-sm", type="submit", title=f"Delete {dist.title}")[
+                htpy.i(".bi.bi-trash"), " Delete"
+            ],
+        ]
+        block.append(htm.div(".mb-3")[delete_form])
 
     title = f"Distribution list for {project} {version}"
     return await template.blank(title, content=block.collect())
 
 
+# The delete_get function can now be removed since we're using inline confirmation
+# But if you want to keep it for direct URL access, you can redirect to the list
+@get.committer("/distribution/delete/<project>/<version>")
+async def delete_get(session: web.Committer, project: str, version: str) -> web.WerkzeugResponse:
+    # Redirect to the list page instead of showing a separate confirmation page
+    return await session.redirect(list_get, project=project, version=version)
+
+
 @get.committer("/distribution/record/<project>/<version>")
 async def record(session: web.Committer, project: str, version: str) -> str:
-    form = await shared.distribution.DistributeForm.create_form(data={"package": project, "version": version})
-    fpv = shared.distribution.FormProjectVersion(form=form, project=project, version=version)
-    return await shared.distribution.record_form_page(fpv)
+    await shared.release_validated(project, version, staging=False)
+
+    block = htm.Block()
```

Comment:
All of this code before `form.render` is duplicated in the `stage` function below, except for the call to `staging` in `html_nav_phase`. We can lift it into a helper function with `staging` parameterised.